File: Week 4 - Memory/code from lecture/swap.c
#include <stdio.h>

void swap(int *a, int *b);

int main(void)
{
    int x = 1;
    int y = 2;

    printf("x is %i\n", x);
    printf("y is %i\n", y);
    printf("Swapping...\n");
    swap(&x, &y); // & means get me the address of x and y variables and pass those in. If it was just x and y then it would be a copy that would be past in
    printf("Swapped!\n");
    printf("x is %i\n", x);
    printf("y is %i\n", y);
}

void swap(int *a, int *b)
{
    int tmp = *a;
    *a = *b;
    *b = tmp;
}

File: Week 8 - Python/miscellanous/swap.py
# But there’s no way to get the pointers in Python, so the only way we can swap values is this:

x = 1
y = 2

print("x is {}".format(x))
print("y is {}".format(y))
print("Swapping...")
x, y = y, x # tuples
# you could do:
# tmp = x
# x = y
# y = temp   # but this is that elegant
print("Swapped.")
print("x is {}".format(x))
print("y is {}".format(y))


# In Python, we can actually swap variables with one line. The left side 
# and right side, x, y, and y, x are both tuples, a data structure with 
# multiple values, and we’re setting the items inside x, y to what the items 
# inside y, x are, which swaps the values.

# A function, too, can return multiple values, so we might need to 
# save them with something like a, b, c, d = foo()

